third_party/proton/csrc/lib/Profiler/CuptiProfiler.cpp

```diff
@@ -209,8 +150,64 @@ std::pair<bool, bool> matchKernelCbId(CUpti_CallbackId cbId) {
 
 } // namespace
 
-void CuptiProfiler::callback(void *userData, CUpti_CallbackDomain domain,
-                             CUpti_CallbackId cbId, const void *cbData) {
+struct CuptiProfiler::CuptiCallback {
```

Comment:
So I named it cuptiCallback because the two classes do have functionality differences.

Functions in `cuptiCallback` are those "callbacks" will intercept cuda runtime/driver calls or process profiling data.

While `CuptiProfiler` is only responsible for invoking cupti apis to register/unregister the profiling functionality

Reply:
Right, my suggestion was to make the two classes have no functionality differences.  Both classes have the same exact API.  The "public class" does nothing other than forward its calls to the "private class".

Reply:
See https://en.cppreference.com/w/cpp/language/pimpl in the ptr_vector_base example.  Notice how ptr_vector_base's public methods do nothing other than call into the pimpl.

Reply:
Oh I see. No problem. I'll make changes

Reply:
@jlebar Updated